editor/components/block-list/insertion-point.js

```diff
@@ -3,36 +3,74 @@
  */
 import { connect } from 'react-redux';
 
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { isUnmodifiedDefaultBlock } from '@wordpress/blocks';
+import { Component } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import {
 	getBlockIndex,
 	getBlockInsertionPoint,
 	isBlockInsertionPointVisible,
-	getBlockCount,
+	getBlock,
+	isTyping,
 } from '../../store/selectors';
+import {
+	insertDefaultBlock,
+	startTyping,
+} from '../../store/actions';
 
-function BlockInsertionPoint( { showInsertionPoint } ) {
-	if ( ! showInsertionPoint ) {
-		return null;
+class BlockInsertionPoint extends Component {
+	constructor() {
+		super( ...arguments );
+		this.onClick = this.onClick.bind( this );
 	}
+	onClick() {
+		const { layout, rootUID, index, ...props } = this.props;
+		props.insertDefaultBlock( { layout }, rootUID, index );
+		props.startTyping();
+	}
+
+	render() {
+		const { showInsertionPoint, showInserter } = this.props;
 
-	return <div className="editor-block-list__insertion-point" />;
+		return (
+			<div className="editor-block-list__insertion-point">
+				{ showInsertionPoint && <div className="editor-block-list__insertion-point-indicator" /> }
+				{ showInserter && (
+					<button
+						className="editor-block-list__insertion-point-inserter"
+						onClick={ this.onClick }
+						aria-label={ __( 'Insert block' ) }
+					/>
+				) }
+			</div>
+		);
+	}
 }
 
 export default connect(
 	( state, { uid, rootUID } ) => {
 		const blockIndex = uid ? getBlockIndex( state, uid, rootUID ) : -1;
-		const insertIndex = blockIndex > -1 ? blockIndex + 1 : getBlockCount( state );
+		const insertIndex = blockIndex + 1;
 		const insertionPoint = getBlockInsertionPoint( state );
+		const block = uid ? getBlock( state, uid ) : null;
 
 		return {
 			showInsertionPoint: (
 				isBlockInsertionPointVisible( state ) &&
 				insertionPoint.index === insertIndex &&
-				insertionPoint.rootUID === rootUID
+				insertionPoint.rootUID === rootUID &&
+				( ! block || ! isUnmodifiedDefaultBlock( block ) )
 			),
+			showInserter: ! isTyping( state ),
```

Comment:
I'm looking to enhance the between-inserter and am unclear the intent with this condition. Why do we only render the inserter button while not typing?

Reply:
Honestly, I don't remember why I added this exactly :)